cli/src/commands/describe.rs

```diff
@@ -133,7 +139,13 @@ pub(crate) fn cmd_describe(
         None
     };
 
-    let commit_descriptions: Vec<(_, _)> = if args.no_edit || shared_description.is_some() {
+    let skip_editor = if args.edit {
+        false
+    } else {
+        args.no_edit || shared_description.is_some()
+    };
```

Comment:
suggestion: After marking `--edit`/`--no-edit` as conflicting as per above, we might want to add an assertion here to ensure that both bools didn't somehow get set.

suggestion: It might be better to use a "positive" form of the name rather than a "negative" form, such as `use_editor` (and then either check `!use_editor` instead of `skip_editor` or reorder conditions as appropriate).

Reply:
I went with `use_editor` in the end, because I think it makes the calculation a bit easier to read (it's now `use_editor = args.edit || (shared_description.is_none() && !args.no_edit)`, which feels roughly like how a human would describe the condition in natural langauge).  The if-condition is now negated, which isn't great, but the alternative would be to swap around the if block and the else block, and the else block is so large that that would be unreadable.

I've added an assertion, plus a comment to clarify what exactly the assertion is there.